src/main/java/org/apache/mesos/scheduler/plan/DefaultInstallStrategy.java

```diff
@@ -18,12 +19,12 @@ public DefaultInstallStrategy(Phase phase) {
     }
 
     @Override
-    public Optional<Block> getCurrentBlock() {
+    public Optional<Block> getCurrentBlock(Collection<String> dirtiedAssets) {
         if (interrupted.get()) {
             return Optional.empty();
         } else {
             for (Block block : phase.getBlocks()) {
-                if (!block.isComplete()) {
+                if (!block.isComplete() && !dirtiedAssets.contains(block.getName())) {
```

Comment:
Yeah, it will work fine when `dirtiedAssets` is empty. 
I think not-doing null check in methods which are not part of public interfaces is okay. But, I think we should exercise caution by validating arguments of methods which are part of public interfaces like `PhaseStrategy`. If we are not doing it at other places in code, maybe we should start doing it.